x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/sp/ServiceProviderCacheSettings.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.idp.saml.sp;
+
+import org.elasticsearch.common.cache.Cache;
+import org.elasticsearch.common.cache.CacheBuilder;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+
+import java.util.List;
+
+/**
+ * Represents standard settings for the ServiceProvider cache(s) in the IdP
+ */
+public final class ServiceProviderCacheSettings {
+    private static final int CACHE_SIZE_DEFAULT = 1000;
```

Comment:
Given that resolving registered and wildcard SPs has different costs, wouldn't it also be plausible that we might need to set the cache size/ttl differently for each ? It might be a premature optimization and I'm fine with a follow-up just wanted to hear your thoughts

Reply:
I don't think it really matters.
Right now, the registered SP cache will be almost completely empty.